sanic/http.py

```diff
@@ -187,12 +187,12 @@ async def http1(self):
                 if self.response:
                     self.response.stream = None
 
-            self.init_for_request()
-
             # Exit and disconnect if no more requests can be taken
             if self.stage is not Stage.IDLE or not self.keep_alive:
                 break
 
+            self.init_for_request()
```

Comment:
@Tronic Do you have any issues with moving this?

Reply:
I gather this should only alter the state in which this returns to protocol, but I don't see any reason for this change or against it. Maybe @ashleysommer could elaborate. Neither `.stage` nor `.keep_alive` is set by `init_for_request`.

Reply:
The point is that the new websockets implementation raises a silent exception when it is the other way around.

Reply:
Ok then.

Reply:
Just to clarify.
One of the things that `init_for_request()` does is it resets `self.upgrade_websocket` to None.
A websocket request will always `break` in the next if statement, because its not KeepAlive.
But the new check in http_protocol for `and not self._http.upgrade_websocket` needs `_http.upgrade_websocket` to still be populated.